README.md

```diff
@@ -83,71 +86,125 @@ There are many common rules for naming and maintaining **Classes** within
+  // Fire the gun.
 }
 ```
 
 **Comments** should be used to describe intention, algorithmic overview, and/or
 logical flow. The ideal comment allows someone other than the author to
 understand a function's intended behavior and general operation.
+
 - Comments should be placed on a separate line, above the code
 - Comment text should begin with an uppercase letter
 - Comment text should end with a period
 - Insert one space between the comment delimiter and the comment text
 
-For **Spacing**, use a single space after a comma between function arguments.
+Example:
+```csharp
+/// This is a comment.
+```
+
+For **Spacing**, use a single space after a comma.
 
 Example: `Console.In.Read(myChar, 0, 1);`
+
 - Do not use a space after the parenthesis and function arguments
 - Do not use spaces between a function name and parenthesis
 - Do not use spaces inside brackets
 
+Class members should be accessed through the `this` keyword, to help tracking the
+scope of each variable used.
+
+Example:
+```csharp
+this.transform.position = this._startPosition;
+```
+
 ### Compiling
+
 All scripts should compile with **zero warnings** and **zero errors**. Warnings
 and errors may quickly cascade into unexpected behavior.
 
 ### Variables
 
 There are many common rules for naming and maintaining **Variables** within
 **Classes**.
+
 - Always use [Pascal-case](#types-of-naming-conventions) for public, internal,
   and protected variables
 - Always use [Camel-case](#types-of-naming-conventions) for private, and local
   variables
 - Always use [Pascal-case](#types-of-naming-conventions) for boolean, and enum
   variables.
-- Always use [Pascal-case](#types-of-naming-conventions) for interfaces
+- Always use [Pascal-case](#types-of-naming-conventions) for interfaces and
+  classes
 - Always prefix private variables with an underscore
-- Always prefix boolean variables with a verb
-- Always prefix interfaces with an `I`
-- All non-boolean variable names must be clear, unambiguous, and descriptive
-  **nouns**
+- Always prefix boolean variables with a verb such as `is`, `has`, ...
+- All variable names must be clear, unambiguous, and descriptive **nouns**
 - Variable names must not be redundant with their context
+- Local variable and function parameter names should not conflict with other
+  variables in the same scope (class variables, global variables, ...)
 - All **Serializable** variables should make use of value ranges if there is
   ever a value that a variable should not be set to
 
-There are many common rules for naming and maintaining **Functions** and **Event Handlers** within **Classes**.
+There are many common rules for naming and maintaining **Functions** and **Event
+Handlers** within **Classes**.
+
 - All function names should start with verbs
 - All function names should be worded in present tense
 - All function names should have context as to what they are doing
 - All event handler names should start with `On`
 
+### Design Principles

```

Comment:
Again that may be a bit much, but I feel the need to include them, at least not as hard rules that people **must** follow, but more as guidelines that people **should** follow as much as possible.

Reply:
I think including design guidelines in the project conventions are not necessary, but a nice addition either way. I appreciate you adding them!